### DISCUSSION TITLE
2021-07-08 Dev Platform

### DISCUSSION THREAD

**Discussion by mikedotexe** [near/community#71] [Archived]
Updates related to Dev Platform team: https://github.com/near/dev-platform
Rust-related board: https://github.com/orgs/near/projects/10
Tooling/libraries board: https://github.com/orgs/near/projects/13

## Done
**Non-Rust**
- Phuong's python code for testing merged to `nearcore`. 🥳 
- Serhii lead a PR-crushing effort for `near-api-js`
- Josh Q. and Serhii finalizing the draft of the CLI proposal, coming to a governance site NEAR you
- Had meeting to discuss what we want out of the Rust SDK book. Some takeaways:
  - Add a prerequisites section
  - Add "Who is this book for" section
  - We liked the direction taken in The Rust Programming Language book. One example: walk through examples with errors
  - (Below, we want the Rust SDK book to be in the **explanation** quadrant and the https://docs.rs/near-sdk/latest/near_sdk site to be the **reference** (image source and guidance from https://documentation.divio.com)
![image](https://user-images.githubusercontent.com/1042667/124950802-98cd2800-dfc7-11eb-9933-d3f615026a21.png)
    - Decided we'd like to avoid documenting rust-specific concepts in the SDK docs.
- Mochi DAO is live and will help alleviate the team from smaller work items. Will be funded more. https://v2.sputnik.fund

**Rust**
- Phuong addressed some formatting issues for the Rust SDK docs
- Austin adding section about transferring NEAR tokens to Rust SDK docs, as well as a page on serialization
- Merged a couple PRs that get us closer to our next major release

## In Progress
- Phuong's work on the Rust SDK's `CurveType` undergoing reviews from Kladov and Austin and may grow to be more ambitious: https://github.com/near/near-sdk-rs/pull/453
- Austin and Phuong working on launching https://sdk-g4yv.onrender.com

**Comment by maxzaver** [comment]
> Serhii lead a PR-crushing effort for near-api-js

@mikedotexe what PRs is it referring to?

**Reply by mikedotexe** [reply]
Here's a capture when Serhii was bringing this up to the team and being on top of it:

> Let's review these PRs:
> 1. https://github.com/near/near-cli/pull/799
> 2. https://github.com/near/near-api-js/pull/623
> 3. https://github.com/near/near-api-js/pull/620/files @Daryl | NEAR , you had some thoughts on this 
> 4. https://github.com/near-examples/near-crossword/pull/1 @Austin Abell | NEAR @Mike Purvis | NEAR 
> 5. We really need to approve final design and merge this PR: https://github.com/near/near-api-js/pull/594 @mehtaphysical 
> 6. https://github.com/near/near-api-js/pull/579 @mehtaphysical @Mike Purvis | NEAR are we ready to merge this one? (we should fix the conflicts first)
> 7. New external contribution: https://github.com/near/near-api-js/pull/625 Do we need this functionality in NAJ?
> 8. https://github.com/near/near-cli/pull/798
> 
> This PR is blocked because we have not released near-api-js for a while: https://github.com/near/near-cli/pull/744
> Should we do that?

**Reply by mehtaphysical** [reply]
For the two I was mentioned on:

5. I agree with @MaximusHaximus comments. We really need to think about `near-api-js` architecture. Before adding a change like this, we should decide what `near-api-js` is supposed to do and settle on a design.
6. We should do this :).

Overall, I think `near-api-js` should be separated into multiple packages in a monorepo. My idea is to make the packages in this doc here: https://www.notion.so/nearprotocol/near-api-js-ideas-c36cd990f1f848a0ac963b915b85c369.

If we agree on this, we should consider putting all our JS/TS libraries in the same monorepo (borsh-js, near-cli, create-near-app). This seems to be the modern conversion, it helps make sure that all repos change together.

**Reply by volovyks** [reply]
@mehtaphysical  Agree on 5 and 6. But let's discuss your last thought at the meeting.
